Reorganize Sway docs and compress some sections (#980)
FuelLabs/sway: docs/src/basics/comments_and_logging.md

# Comments and Logging

## Comments

> Comments in Sway start with two slashes and continue until the end of the line. For comments that extend beyond a single line, you’ll need to include // on each line.

Reviewer: ```suggestion
Comments in Sway start with two slashes and continue until the end of the line. For comments that extend beyond a single line, you’ll need to include `//` on each line.
```

> Comments in Sway start with two slashes and continue until the end of the line. For comments that extend beyond a single line, you’ll need to include `//` on each line.

Reviewer: Search-and-replace all smart quotes into dumb quotes.
```suggestion
Comments in Sway start with two slashes and continue until the end of the line. For comments that extend beyond a single line, you'll need to include `//` on each line.
```

```sway
// hello world
```

```sway
// let's make a couple of lines
// commented.
```

You can also place comments at the ends of lines containing code.

```sway
fn main() {
    let baz = 8; // Eight is a lucky number
}
```

## Logging

> To log integers to stdout, you can use the `log_u64`, `log_u32`, `log_u16`, or `log_u8` functions from the `lib-std` library.

Reviewer: Logging isn't like `console.log`. See https://blog.chain.link/events-and-logging-in-solidity/ for more info. This deserves its own page tbh, but can always be left to a subsequent PR.
```suggestion
To log integers, you can use the `log_u64`, `log_u32`, `log_u16`, or `log_u8` functions from the `lib-std` library.
```

```sway

> use std::chain::log_u64

Reviewer: !

This is why code examples should all be included from real Sway files that are compiled as part of CI 😂
```suggestion
use std::chain::log_u64;
```
